### ISSUE TITLE
Headless wiki examples don't work

### ISSUE THREAD

**Issue by randyzwitch** [vega/vega#356]
I think this is just an issue of reorganizing the libraries between v1 and v2 (I already modified the Wiki in a few places).

Running the following wiki example gives me an error about d3:

```
HQSML-146932:vega rzwitc200$ bin/vg2png test/spec/bar.json bar.png
module.js:338
    throw err;
          ^
Error: Cannot find module 'd3'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/rzwitc200/.julia/v0.4/Vega/deps/vega/bin/vg2png:15:10)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
```

**Comment by arvind** [comment]
Hi Randy,

Have you run `npm install` within your Vega directory? That should install
all dependencies you'll need to run Vega within node.js/headless, including
D3.

On Monday, September 14, 2015, Randy Zwitch <notifications@github.com>
wrote:

> I think this is just an issue of reorganizing the libraries between v1 and
> v2 (I already modified the Wiki in a few places).
>
> Running the following wiki example gives me an error about d3:
>
> HQSML-146932:vega rzwitc200$ bin/vg2png test/spec/bar.json bar.png
> module.js:338
>     throw err;
>           ^
> Error: Cannot find module 'd3'
>     at Function.Module._resolveFilename (module.js:336:15)
>     at Function.Module._load (module.js:278:25)
>     at Module.require (module.js:365:17)
>     at require (module.js:384:17)
>     at Object.<anonymous> (/Users/rzwitc200/.julia/v0.4/Vega/deps/vega/bin/vg2png:15:10)
>     at Module._compile (module.js:460:26)
>     at Object.Module._extensions..js (module.js:478:10)
>     at Module.load (module.js:355:32)
>     at Function.Module._load (module.js:310:12)
>     at Function.Module.runMain (module.js:501:10)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vega/vega/issues/356>.
>


-- 
-Arvind
-----
Arvind Satyanarayan <http://arvindsatya.com/>

**Comment by randyzwitch** [comment]
I didn't, thank you, that fixed the d3 issue.

Running `bin/vg2png test/spec/bar.json bar.png` gives the following picture:

![bar](https://cloud.githubusercontent.com/assets/2762787/9854217/a42c410e-5ad5-11e5-83bc-f66e453171a5.png)

**Comment by jheer** [comment]
Hi Randy, unfortunately I can not replicate the issue. The png renders fine for me on Mac OS X with the identical command. Can you include your operating system info and (ideally) also your installed version of Cairo (the native renderer used by node-canvas)?

**Comment by arvind** [comment]
Ah yes, I remember this issue. I believe it has to do with [a bug](
https://bugs.freedesktop.org/show_bug.cgi?id=84324#c13) with Cairo and OS X
Yosemite. It looks like the bug has been marked as resolved, so I'd try the
latest version of Cairo. If you're still not having any luck, I have
headless rendering reliably working with Yosemite and Cairo 0.12.x.

On Monday, September 14, 2015, Randy Zwitch <notifications@github.com>
wrote:

> I didn't, thank you, that fixed the d3 issue.
>
> Running bin/vg2png test/spec/bar.json bar.png gives the following picture:
>
> [image: bar]
> <https://cloud.githubusercontent.com/assets/2762787/9854217/a42c410e-5ad5-11e5-83bc-f66e453171a5.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vega/vega/issues/356#issuecomment-140119902>.
>


-- 
-Arvind
-----
Arvind Satyanarayan <http://arvindsatya.com/>

**Comment by randyzwitch** [comment]
Perfect, thanks for your help!

**Comment by randyzwitch** [comment]
@jheer, I am using `cairo 1.14.2_1` from homebrew, which appears to be the newest one I can get from homebrew.

**Comment by jheer** [comment]
That may be part of the issue. FWIW, I'm on Cairo `1.12.16_1`.

**Comment by jpocom** [comment]
I had the same problem with cairo 1.14.2 from homebrew. 
I patched "cairo 1.14.2"  with https://bugs.freedesktop.org/attachment.cgi?id=114401 and it works.
